pkg/trace/agent/obfuscate_test.go

```diff
@@ -50,6 +52,84 @@ func TestMetaHook(t *testing.T) {
 	}
 }
 
+func TestAppSecMetaStructHook(t *testing.T) {
+	cco := newCreditCardsObfuscator(config.CreditCardsConfig{Enabled: true})
+	defer cco.Stop()
+
+	t.Run("no creditcards", func(t *testing.T) {
+
+		appsec := pb.AppSecStruct{Triggers: []*pb.AppSecTrigger{{
+			Rule: &pb.AppSecRuleTrigger{Id: "ua-000-01", Name: "Arachni"},
+			RuleMatches: []*pb.AppSecRuleMatch{{
+				Operator:      "regex_match",
+				OperatorValue: "Arachni",
+				Parameters: []*pb.AppSecRuleParameter{
+					{
+						Address:   "http.request.headers",
+						Value:     "Arachni/v1",
+						Highlight: []string{"Arachni"},
+					},
+				},
+			}},
+		}}}
+
+		appsecb := []byte{}
+		appsecb, err := appsec.MarshalMsg(appsecb)
+		if err != nil {
+			t.Fatalf("couldn't marshal appsec struct: %v", err)
+		}
+
+		assert.Equal(t, appsecb, cco.MetaStructHook("appsec", appsecb))
+	})
+
+	t.Run("with creditcards", func(t *testing.T) {
+
+		appsec := pb.AppSecStruct{Triggers: []*pb.AppSecTrigger{{
+			Rule: &pb.AppSecRuleTrigger{Id: "ua-000-01", Name: "5105-1051-0510-5100"},
+			RuleMatches: []*pb.AppSecRuleMatch{{
+				Operator:      "regex_match",
+				OperatorValue: "Arachni",
+				Parameters: []*pb.AppSecRuleParameter{
+					{
+						Address:   "http.request.headers",
+						Value:     "5105-1051-0510-5100",
+						Highlight: []string{"5105-1051-0510-5100"},
+					},
+				},
+			}},
+		}}}
+
+		appsecb := []byte{}
+		appsecb, err := appsec.MarshalMsg(appsecb)
+		if err != nil {
+			t.Fatalf("couldn't marshal appsec struct: %v", err)
+		}
+
+		obfuscated := cco.MetaStructHook("appsec", appsecb)
```

Comment:
```suggestion
		v := cco.MetaStructHook("appsec", appsecb)
```
I think `v` is more appropriate here, since we are returning a new **v**alue for the meta map. Furthermore, the MetaStructHook is a concept agnostic of obfuscation, so it could be anything, right? Granted that in this context you are calling it on `cco` so it's more clear.

Not feeling strongly on this. If you prefer it, leave it.